src/PromTracker.mo

```diff
@@ -0,0 +1,291 @@
+import AssocList "mo:base/AssocList";
+import Cycles "mo:base/ExperimentalCycles";
+import Int "mo:base/Int";
+import Iter "mo:base/Iter";
+import Nat "mo:base/Nat";
+import Nat64 "mo:base/Nat64";
+import Prim "mo:prim";
+import StableMemory "mo:base/ExperimentalStableMemory";
+import Time "mo:base/Time";
+import Text "mo:base/Text";
+
+import Vec "mo:vector";
+
+module {
+
+  type StableDataItem = { #counter : Nat; #gauge : (Nat, Nat) };
+  public type StableData = AssocList.AssocList<Text, StableDataItem>;
+
+  type ValueRefMixin = { value : () -> Nat; remove : () -> () };
+  /// A reference to pull value
+  public type PullValueRef = ValueRefMixin;
+  /// A reference to accumulator value
+  public type AccumulatorValueRef = ValueRefMixin and {
+    set : (x : Nat) -> ();
+    add : (x : Nat) -> ();
+  };
+  /// A reference to gauge value
+  public type GaugeValueRef = ValueRefMixin and { update : (x : Nat) -> () };
```

Comment:
I would call the "interfaces" instead of "references". Maybe `GaugeInterface, CounterInterface, PullValueInterface`.

Not sure if it's worth combining them with `and` from a base type. Might be easier to just repeat. If we use a base type I would use "base" instead of "mixin" in the name.

Reply:
Done